### PR TITLE
FOGL-3064 notification plugins API update added

### PR DESCRIPTION
**Pull request by ashish-jabble**
Signed-off-by: ashish-jabble <ashish@dianomic.com>

### CODE REVIEW

**Review by MarkRiddoch — DISMISSED** [review]

File: python/foglamp/services/core/api/plugins/update.py [hunk 1]
```diff
@@ -39,48 +40,76 @@
...
+                config_mgr = ConfigurationManager(storage_client)
+                all_notifications = await config_mgr._read_all_child_category_names("Notifications")
+                for notification in all_notifications:
+                    notification_config = await config_mgr._read_category_val(notification['child'])
+                    notification_name = notification_config['name']['value']
+                    channel = notification_config['channel']['value']
+                    rule = notification_config['rule']['value']
+                    is_enabled = True if notification_config['enable']['value'] == 'true' else False
+                    if (channel == name and is_enabled) or (rule == name and is_enabled):
+                        _logger.warning("{} notification instance is disabled as {} {} plugin is updating..".format(
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/plugins/update.py]: `Disabling {} notification instance, as {} {} plugin is updating...`

File: python/foglamp/services/core/api/plugins/update.py [hunk 2]
```diff
@@ -39,48 +40,76 @@
...
 
-        # Tracked plugins from asset tracker
-        tracked_plugins = await _get_plugin_and_sch_name_from_asset_tracker(_type)
         sch_list = []
-        filters_used_by = []
-        if _type == 'filter':
-            # In case of filter, for asset_tracker table we are inserting filter category_name in plugin column
-            # instead of filter plugin name by Design
-            # Hence below query is required to get actual plugin name from filters table
+        notify_list = []
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/plugins/update.py]: this should be `notification_instances_list`

File: python/foglamp/services/core/api/plugins/update.py [hunk 3]
```diff
@@ -39,48 +40,76 @@
...
+        if _type in ['notificationDelivery', 'notificationRule']:
+            # Check Notification service is enabled or not
+            payload = PayloadBuilder().SELECT("id", "enabled", "schedule_name").WHERE(['process_name', '=',
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/plugins/update.py:65]: can not there be multiple schedules of notification_c, (may be disabled)? This is important to fix as we are checking sch_info[0].

**Review by praveen-garg — CHANGES_REQUESTED** [review]

File: python/foglamp/services/core/api/plugins/update.py (hunk 3 above)
Comment (ashish-jabble) [reply python/foglamp/services/core/api/plugins/update.py:65]: Nope, we have added a restriction of notification service with one instance only when hit  add_service endpoint

Comment (praveen-garg) [reply python/foglamp/services/core/api/plugins/update.py:65]: Okay

**Review by praveen-garg — APPROVED** [review]
lgtm!

**Pull request merged**